src/main/java/org/broadinstitute/consent/http/resources/NihAccountResource.java

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.consent.http.resources;
+
+import org.broadinstitute.consent.http.models.NIHUserAccount;
+import org.broadinstitute.consent.http.service.NihAuthApi;
+
+import javax.annotation.security.RolesAllowed;
+
+import javax.ws.rs.Path;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.core.Response;
+
+@Path("api/nih-register/")
+
+public class NihAccountResource extends Resource {
+
+    private NihAuthApi nihAuthApi;
+
+    public NihAccountResource(NihAuthApi nihAuthApi) {
+        this.nihAuthApi = nihAuthApi;
+    }
+
+    @POST
+    @Path("{userId}")
+    @Produces("application/json")
+    @RolesAllowed("RESEARCHER")
+
+    public Response registerResearcher(@PathParam("userId") Integer userId, NIHUserAccount nihAccount) {
+        try {
+            return Response.status(Response.Status.OK).entity(nihAuthApi.authenticateNih(nihAccount, userId)).build();
```

Comment:
This looks susceptible to hacking to me. If Researcher A is able to glean the id of another researcher, then Researcher A could authenticate and then set or delete the NIH status of a different user. I think we need some validation that the researcher can only add or remove their own ID. 

So, I don't think this endpoint should take a userId parameter. It should deduce that from the authenticated user. Same with the delete.

Reply:
With this new change, these endpoints will only work for a logged user profile.